### ISSUE TITLE
sway window#waybar.solo triggers on a split

### ISSUE THREAD

**Issue by ralsei** [Alexays/Waybar#671]
To reproduce, use CSS like:

```css
window {
  color:      #c2c2b0;
  background: rgba(34, 34, 34, 0.00);
}
window#waybar.solo {
  color:      #c2c2b0;
  background: rgba(34, 34, 34, 0.85);
}
```

Then [in Sway] open two windows, create a vertical split, and open and close a window. The style for `waybar.solo` triggers even though there are two workspaces.

Example provided at https://qtp2t.club/pub/waybar.mp4. Using Sway 1.4 and Waybar 0.9.0 under Void Linux.

**Comment by narilth** [comment]
I have a fix for this issue, but I'm not sure how to submit it because this is my first open-source contribution. Please tell me how I can do that.

---

From what I understand, this happens because Sway's list of nodes in a workspace isn't flat. On a workspace with two windows, you can have the following tree.

```json
{
    "type": "workspace",
    "nodes": [
        {"nodes": ["window0"]},
        {"nodes": ["window1"]}
    ]
}
```

To count the number of nodes in a workspace, I count all the leaf nodes of the workspace node that contains the focused window.

**Comment by narilth** [comment]
I think I figured out how to submit the change. I made the pull request narilth:sway-solo-css-fix

**Comment by Alexays** [comment]
Should be fixed with #946
